src/crafting_gui.cpp

```diff
@@ -109,8 +109,17 @@ int print_items( const recipe &r, const catacurses::window &w, int ypos, int xpo
 
     mvwprintz( w, ypos++, xpos, col, _( "Byproducts:" ) );
     for( const auto &bp : r.byproducts ) {
-        mvwprintz( w, ypos++, xpos, col, _( "> %d %s" ), bp.second * batch,
-                   item::nname( bp.first ).c_str() );
+        int amount = bp.second * batch;
+        std::string desc;
+        if( item::count_by_charges( bp.first ) ) {
+            amount *= item::find_type( bp.first )->charges_default();
+            desc = string_format( _( "> %s (%d)" ),
+                                  item::nname( bp.first, 1 ).c_str(), amount );
```

Comment:
Both `count_by_charges()` and `nname()` call `find_type()` under the hood. It could be changed to something like:
```c++
const auto t = item::find_type( bp.first );

if( t->count_by_charges() ) {
    amount *= t->default_charges();
    ...
}
```